File: CSharp/OpenTraceability/Models/Events/SensorReport.cs
using OpenTraceability.Interfaces;
using OpenTraceability.Utility;
using OpenTraceability.Utility.Attributes;

namespace OpenTraceability.Models.Events
{
    public class SensorReport
    {
        [OpenTraceabilityJson("time")]
        [OpenTraceability("@time")]
        public DateTimeOffset? TimeStamp { get; set; }

        [OpenTraceabilityJson("type")]
        [OpenTraceability("@type")]
        public Uri Type { get; set; }

        [OpenTraceabilityJson("value")]
        [OpenTraceability("@value")]
        public double? Value { get; internal set; }

        [OpenTraceabilityJson("stringValue")]
        [OpenTraceability("@stringValue")]
        public string StringValue { get; internal set; }

        [OpenTraceabilityJson("booleanValue")]
        [OpenTraceability("@booleanValue")]
        public bool? BooleanValue { get; internal set; }

        [OpenTraceabilityJson("hexBinaryValue")]
        [OpenTraceability("@hexBinaryValue")]
        public string HexBinaryValue { get; internal set; }

        [OpenTraceabilityJson("uriValue")]
        [OpenTraceability("@uriValue")]
        public Uri URIValue { get; internal set; }

        [OpenTraceabilityJson("uom")]
        [OpenTraceability("@uom")]
        public UOM UOM { get; internal set; }

        [OpenTraceabilityJson("component")]
        [OpenTraceability("@component")]
        public Uri Component { get; internal set; }

        [OpenTraceabilityJson("minValue")]
        [OpenTraceability("@minValue")]
        public double? MinValue { get; internal set; }

        [OpenTraceabilityJson("maxValue")]
        [OpenTraceability("@maxValue")]
        public double? MaxValue { get; internal set; }

        [OpenTraceabilityJson("sDev")]
        [OpenTraceability("@sDev")]
        public double? SDev { get; internal set; }

        [OpenTraceabilityJson("chemicalSubstance")]
        [OpenTraceability("@chemicalSubstance")]
        public Uri ChemicalSubstance { get; internal set; }

        [OpenTraceabilityJson("microorganism")]
        [OpenTraceability("@microorganism")]
        public Uri MicroOrganism { get; internal set; }

        [OpenTraceabilityJson("deviceID")]
        [OpenTraceability("@deviceID")]
        public Uri DeviceID { get; internal set; }

        [OpenTraceabilityJson("rawData")]
        [OpenTraceability("@rawData")]
        public Uri RawData { get; internal set; }

        [OpenTraceabilityJson("meanValue")]
        [OpenTraceability("@meanValue")]
        public double? MeanValue { get; internal set; }

        [OpenTraceabilityJson("percRank")]
        [OpenTraceability("@percRank")]
        public double? PercentageRank { get; internal set; }

        [OpenTraceabilityJson("dataProcessingMethod")]
        [OpenTraceability("@dataProcessingMethod")]
        public Uri DataProcessingMethod { get; internal set; }

        [OpenTraceabilityJson("coordinateReferenceSystem")]
        [OpenTraceability("@coordinateReferenceSystem")]
        public Uri CoordinateReferenceSystem { get; internal set; }

        [OpenTraceabilityJson("exception")]
        [OpenTraceability("@exception")]
        public Uri Exception { get; internal set; }

        [OpenTraceabilityJson("percValue")]
        [OpenTraceability("@percValue")]
        public double? PercentageValue { get; internal set; }

        [OpenTraceabilityJson("deviceMetadata")]
        [OpenTraceability("@deviceMetadata")]
        public Uri DeviceMetadata { get; internal set; }

        [OpenTraceabilityExtensionAttributes]
        public List<IEventKDE> ExtensionAttributes { get; internal set; } = new List<IEventKDE>();
    }
}

File: JDK11/opentraceability/src/main/java/opentraceability/models/events/sensorreport.java
package opentraceability.models.events;

import opentraceability.interfaces.IEventKDE;
import opentraceability.utility.UOM;
import opentraceability.utility.attributes.OpenTraceabilityAttribute;
import opentraceability.utility.attributes.OpenTraceabilityExtensionAttributesAttribute;
import opentraceability.utility.attributes.OpenTraceabilityJsonAttribute;

import java.net.URI;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;

public class SensorReport {
    @OpenTraceabilityJsonAttribute("time")
    @OpenTraceabilityAttribute("", "@time")
    public OffsetDateTime timeStamp = null;

    @OpenTraceabilityJsonAttribute("type")
    @OpenTraceabilityAttribute("", "@type")
    public URI type = null;

    @OpenTraceabilityJsonAttribute("value")
    @OpenTraceabilityAttribute("", "@value")
    public Double value = null;

    @OpenTraceabilityJsonAttribute("stringValue")
    @OpenTraceabilityAttribute("", "@stringValue")
    public String stringValue = null;

    @OpenTraceabilityJsonAttribute("booleanValue")
    @OpenTraceabilityAttribute("", "@booleanValue")
    public Boolean booleanValue = null;

    @OpenTraceabilityJsonAttribute("hexBinaryValue")
    @OpenTraceabilityAttribute("", "@hexBinaryValue")
    public String hexBinaryValue = null;

    @OpenTraceabilityJsonAttribute("uriValue")
    @OpenTraceabilityAttribute("", "@uriValue")
    public URI uriValue = null;

    @OpenTraceabilityJsonAttribute("uom")
    @OpenTraceabilityAttribute("", "@uom")
    public UOM uom = new UOM();

    @OpenTraceabilityJsonAttribute("component")
    @OpenTraceabilityAttribute("", "@component")
    public URI component = null;

    @OpenTraceabilityJsonAttribute("minValue")
    @OpenTraceabilityAttribute("", "@minValue")
    public Double minValue = null;

    @OpenTraceabilityJsonAttribute("maxValue")
    @OpenTraceabilityAttribute("", "@maxValue")
    public Double maxValue = null;

    @OpenTraceabilityJsonAttribute("sDev")
    @OpenTraceabilityAttribute("", "@sDev")
    public Double sDev = null;

    @OpenTraceabilityJsonAttribute("chemicalSubstance")
    @OpenTraceabilityAttribute("", "@chemicalSubstance")
    public URI chemicalSubstance = null;

    @OpenTraceabilityJsonAttribute("microorganism")
    @OpenTraceabilityAttribute("", "@microorganism")
    public URI microOrganism = null;

    @OpenTraceabilityJsonAttribute("deviceID")
    @OpenTraceabilityAttribute("", "@deviceID")
    public URI deviceID = null;

    @OpenTraceabilityJsonAttribute("rawData")
    @OpenTraceabilityAttribute("", "@rawData")
    public URI rawData = null;

    @OpenTraceabilityJsonAttribute("meanValue")
    @OpenTraceabilityAttribute("", "@meanValue")
    public Double meanValue = null;

    @OpenTraceabilityJsonAttribute("percRank")
    @OpenTraceabilityAttribute("", "@percRank")
    public Double percentageRank = null;

    @OpenTraceabilityJsonAttribute("dataProcessingMethod")
    @OpenTraceabilityAttribute("", "@dataProcessingMethod")
    public URI dataProcessingMethod = null;

    @OpenTraceabilityJsonAttribute("coordinateReferenceSystem")
    @OpenTraceabilityAttribute("", "@coordinateReferenceSystem")
    public URI coordinateReferenceSystem = null;

    @OpenTraceabilityJsonAttribute("exception")
    @OpenTraceabilityAttribute("", "@exception")
    public URI exception = null;

    @OpenTraceabilityJsonAttribute("percValue")
    @OpenTraceabilityAttribute("", "@percValue")
    public Double percentageValue = null;

    @OpenTraceabilityJsonAttribute("deviceMetadata")
    @OpenTraceabilityAttribute("", "@deviceMetadata")
    public URI deviceMetadata = null;

    @OpenTraceabilityExtensionAttributesAttribute
    public List<IEventKDE> extensionAttributes = new ArrayList<>();
}